projects/observability/src/pages/explorer/explorer-dashboard-builder.ts

```diff
@@ -565,9 +570,14 @@ export class ExplorerDashboardBuilder {
     }
   }
 
-  private getUserRequestedNonDefaultColumns(
+  private removeColumnsDuplicated(columns: ModelJson[]): ModelJson[] {
+    return uniqBy(columns, column => (column?.value as AttributeValue).attribute ?? '');
```

Comment:
Rather than casting here, change the declared return types on your two methods (getGeneratedTableColumns, getDefaultTableColumns) to more descriptively describe their results. That's the whole benefit of using types - if your data doesn't conform to that, then this isn't a safe transform to do.

Reply:
I improved it a bit, not possible to change return type of getGeneratedTableColumns because is protected

Reply:
Why does it being protected stop us? It just means any child classes would need to be similarly updated. Not ideal, but if it's the best solution that's never stopped us before.

Reply:
Fixed